### PR TITLE
✨ [Amp story] [Page attachments] [Outlink] Opening Transition Element and Animation

### LINKED ISSUE #33812: [Page attachments] [Outlink] Opening Transition State
Transition state (showing link address) already exists when outlinks are clicked, but now need to show when swiped as well. See mock:

Screen Shot 2021-04-14 at 11 14 20 AM

@ampproject/wg-stories

### PR DESCRIPTION
**Pull request by processprocess**
Context / fixes #33812

- Theme link preview state
- Share SVG code between outlink button and preview state
- RTL support
- Localized `Opening` string

Apr-23-2021 12-01-55

[demo](https://philipbell-stamp-ui.web.app/examples/amp-story/attachment.html)

### CODE REVIEW

File: extensions/amp-story/1.0/amp-story-page-attachment.css [hunk 1]
```diff
@@ -83,3 +83,113 @@
...
+  content: "" !important;
+  position: absolute !important;
+  width: 100% !important;
+  height: 2px !important;
+  color: black !important;
+  bottom: 0 !important;
+  left: 0 !important;
+  background-color: var(--i-amphtml-outlink-cta-text-color) !important;
+  opacity: .6 !important;
+  transform-origin: left !important;
```
Comment (mszylkowski) [inline extensions/amp-story/1.0/amp-story-page-attachment.css:178]: Nit: rtl

File: extensions/amp-story/1.0/amp-story-page-attachment.css [hunk 2]
```diff
@@ -83,3 +83,113 @@
...
+    transform: scaleX(0)
+  }
+  100% {
+    transform: scaleX(1)
+  }
+}
+
+amp-story:not([desktop]) .i-amphtml-amp-story-page-attachment-ui-v2.i-amphtml-story-page-attachment-remote .i-amphtml-story-draggable-drawer-spacer {
+    /* Spacer is not needed for remote attachment.  */
+    height: 0 !important;
```
Comment (mszylkowski) [inline extensions/amp-story/1.0/amp-story-page-attachment.css:193]: Nit: Not 100% sure what this refers to, but does `display: none` work better?

File: extensions/amp-story/1.0/amp-story-page-attachment.css [hunk 3]
```diff
@@ -83,3 +83,113 @@
...
+  flex-shrink: 0 !important;
+  fill: var(--i-amphtml-outlink-cta-text-color) !important;
+  background: none !important;
+}
+
+.i-amphtml-amp-story-page-attachment-ui-v2.i-amphtml-story-page-attachment-remote.i-amphtml-story-draggable-drawer-open:before {
+  content: "" !important;
+  position: absolute !important;
+  width: 100% !important;
+  height: 2px !important;
```
Comment (mszylkowski) [inline extensions/amp-story/1.0/amp-story-page-attachment.css:172]: Note: Is hard to see currently, specially when the color matches the chrome UI (left a screenshot below of my phone)

**Review by mszylkowski — COMMENTED** [review]
Good progress! Left a couple of nits

File: extensions/amp-story/1.0/amp-story-page-attachment.css (hunk 3 above)
Comment (processprocess) [reply extensions/amp-story/1.0/amp-story-page-attachment.css]: Double checked the mocks and it's actually 4px which helps visibility a lot. Demo is updated.

File: extensions/amp-story/1.0/amp-story-page-attachment.js [hunk 4]
```diff
@@ -185,6 +193,58 @@ export class AmpStoryPageAttachment extends DraggableDrawer {
...
+    this.setOpenThreshold_(150 /* pixels */);
+    this.setDragCap_(56 /* pixels */);
+
+    this.headerEl_.classList.add(
+      'i-amphtml-story-draggable-drawer-header-attachment-remote'
+    );
+    this.element.classList.add('i-amphtml-story-page-attachment-remote');
+
+    const link = htmlFor(this.element)`
+      <a class="i-amphtml-story-page-attachment-remote-content" target="_blank">
```
[the remark this answers is not in the document]
Comment (processprocess) [reply extensions/amp-story/1.0/amp-story-page-attachment.js:210]: It's not intended. Thanks for catching this. 

To prevent this we could put `pointer-events: none` on the element:
https://github.com/ampproject/amphtml/pull/33982/files#diff-23ab4d9dc3d70d03d13e3783ce2f7714ba4945fc8079aebd439dd31336dad9d6R115

And enable pointer events when in vertical rendering:
https://github.com/ampproject/amphtml/pull/33982/files#diff-83cbbde09142d0f60e6388a9a7a0bece2ea540fcad9ee699cb703c04c1cedf3fR54-R57

I don't have much context on vertical rendering and how bots are reading our pages. 
I want to be sure this wouldn't introduce any side effects.
@newmuis @gmajoulet

[the remark this answers is not in the document]
Comment (processprocess) [reply extensions/amp-story/1.0/amp-story-page-attachment.js:208]: This looks like it was done before the CTA button was an anchor element.
We can likely make this a div but we will need to be sure bots can crawl the CTA button.

I reverted the patch for this in this PR so we can follow up on it separately. 
I created #34015 for tracking this.

**Review by gmajoulet — APPROVED** [review]

File: extensions/amp-story/1.0/amp-story-page.js [hunk 5]
```diff
@@ -1775,7 +1775,10 @@ export class AmpStoryPage extends AMP.BaseElement {
       container.classList.add('i-amphtml-story-page-open-attachment-host');
       container.setAttribute('role', 'button');
 
-      container.addEventListener('click', () => this.openAttachment());
+      container.addEventListener('click', (e) => {
+        e.preventDefault();
```
Comment (Enriqe) [inline extensions/amp-story/1.0/amp-story-page.js:1779]: sorry, why is this needed?

File: extensions/amp-story/1.0/amp-story-page-attachment.js [hunk 6]
```diff
@@ -185,6 +191,56 @@ export class AmpStoryPageAttachment extends DraggableDrawer {
       );
   }
 
+  /**
+   * Builds remote V2 page attachment's UI.
+   * @private
+   */
+  buildRemoteV2_() {
+    this.setOpenThreshold_(150 /* pixels */);
```
Comment (Enriqe) [inline extensions/amp-story/1.0/amp-story-page-attachment.js:199]: nit: these should be top-level consts

File: extensions/amp-story/1.0/amp-story-page.js (hunk 5 above)
Comment (processprocess) [reply extensions/amp-story/1.0/amp-story-page.js:1780]: Good call. Added a comment.

**Comment by processprocess** [comment]
cc @ jridgewell for OWNERS on i18n.

File: extensions/amp-story/1.0/amp-story-page-attachment.js [hunk 7]
```diff
@@ -24,14 +24,35 @@ import {closest, removeElement} from '../../../src/dom';
...
 import {triggerClickFromLightDom} from './utils';
 
 /** @const {string} */
 const DARK_THEME_CLASS = 'i-amphtml-story-draggable-drawer-theme-dark';
 
+/**
+ * Distance in pixels to swipe before opening attachment.
+ * @const {number}
+ */
+const OPEN_THRESHOLD = 150;
```
Comment (Enriqe) [inline extensions/amp-story/1.0/amp-story-page-attachment.js:41]: nit, here and below:

```suggestion
const OPEN_THRESHOLD_PX = 150;
```

File: extensions/amp-story/1.0/amp-story-page-attachment.js [hunk 8]
```diff
@@ -282,23 +357,33 @@ export class AmpStoryPageAttachment extends DraggableDrawer {
...
+      .shadowRoot.querySelector('a.i-amphtml-story-page-open-attachment');
+
+    const isMobileUI =
+      this.storeService_.get(StateProperty.UI_STATE) === UIType.MOBILE;
+    // Shows outlink url preview on mobile only.
+    if (!isMobileUI) {
+      triggerClickFromLightDom(clickTarget, this.element);
+    } else {
+      const animationEl = this.win.document.createElement('div');
+      animationEl.classList.add('i-amphtml-story-page-attachment-expand');
```
Comment (Enriqe) [inline extensions/amp-story/1.0/amp-story-page-attachment.js:371]: is there a reason why this is not inside the `mutate` phase below? I'm guessing it messes up the animation but just wanted to make sure.

**Review by Enriqe — APPROVED** [review]

**Review by jridgewell — APPROVED** [review]
Please add wg-stories as an owner of the file.
